model/dependencies.go

```diff
@@ -90,8 +99,24 @@ func (di *dependencyIncluder) handle(pair TVPair) (bool, error) {
 
 	// queue up all dependencies for recursive inclusion
 	deps := di.expandDependencies(pair, bvt.DependsOn)
+
+	// If this function is invoked from generate.tasks, calculate all variants
+	// that need to be included as dependencies, but are not in the generated project.
+	// If all the task / variant pairs that spawn these dependencies are inactive, we
+	// also mark this newly generated variant as inactive.
+	pairSpecifiesActivation := activationInfo != nil && activationInfo.taskOrVariantHasSpecificActivation(pair.Variant, pair.TaskName)
 	for _, dep := range deps {
-		ok, err := di.handle(dep)
+		if activationInfo != nil && !activationInfo.variantExistsInGeneratedProject(dep.Variant) {
+			// If the new variant has not yet been added to deactivateGeneratedVariantTasks, or if the
+			// original pair needs to be active, we update deactivateGeneratedVariantTasks.
+			// We ultimately will only deactivate new variants where deactivateGeneratedVariantTasks[variant] = true.
+			// If deactivateGeneratedVariantTasks[variant] = false it signifies that there was at least
+			// one pair that depends on this new variant being active - so we cannot deactivate it.
+			if _, ok := di.deactivateGeneratedVariantTasks[dep.Variant]; !ok || !pairSpecifiesActivation {
+				di.deactivateGeneratedVariantTasks[dep.Variant] = pairSpecifiesActivation
```

Comment:
The name deactivateGeneratedVariantTasks is a bit misleading, since the "tasks" implies that we're storing task names, not variants. We also ultimately wrap this into activationVariants it looks like, so maybe instead this should be generatedActivationVariants?

Reply:
I've renamed to `deactivateGeneratedDep` since really the variable represents if a dependency should be added to the activation info's list of tasks with specific activation (i.e. the list to deactivate)